pallets/identity/src/auth.rs

```diff
@@ -157,9 +158,17 @@ impl<T: Config> Module<T> {
         auth_id: &u64,
     ) -> Option<Authorization<T::AccountId, T::Moment>> {
         Self::authorizations(target, *auth_id).filter(|auth| {
-            auth.expiry
+            // check if count is zero
+            let count_result = auth.count.eq(&0u32);
```

Comment:
Since the count  check is cheap, do that first and return.  That will save having to do the expiry check, which is more expensive.
```suggestion
            // check if count is zero
            if auth.count == &0u32 {
                return true;
            }
```